src/modules/wallet/utils.ts

```diff
@@ -250,3 +246,20 @@ export function getAddEthereumChainParameters(
       }
   }
 }
+
+export async function getProviderChainId(provider: Provider) {
+  const providerChainId = (await provider.request({
+    method: 'eth_chainId',
+    params: []
+  })) as string | number
+
+  let chainId: number
+
+  if (typeof providerChainId === 'string') {
+    chainId = parseInt(providerChainId as string, 16)
+  } else {
+    chainId = providerChainId
+  }
```

Comment:
Do you think it might be useful to add a comment here explaining what's going on? Or maybe just a JSDoc on the getProviderChainId function